Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/VesuvioDiffractionReduction.py

```diff
@@ -94,11 +92,26 @@ def PyExec(self):
         load_opts = dict()
         load_opts['Mode'] = 'FoilOut'
         load_opts['InstrumentParFile'] = self._par_filename
+        # Load monitors as True so monitors will not be loaded separately in LoadVesuvio
+        load_opts['LoadMonitors'] = True
 
         prog_reporter = Progress(self, start=0.0, end=1.0, nreports=1)
 
         prog_reporter.report("Loading Files")
 
+        # Split up runs as LoadVesuvio sums multiple runs
+        input_files = self._data_files
+
+        for run in input_files:
+            if "-" in run:
```

Comment:
You can get Mantid to generate the numbers for you instead of trying to parse it yourself:
```python
    try:
        number_generator = kernel.IntArrayProperty('array_generator', run)
        self._data_files = number_generator.value.tolist()
    except RuntimeError:
        raise RuntimeError("Could not generate run numbers from this input: " + str(run))
```